File: storage/src/androidTest/java/pl/expensive/storage/Injector.java
package pl.expensive.storage;

import android.content.Context;
import android.support.test.InstrumentationRegistry;

class Injector {
    static Context provideContext() {
        return InstrumentationRegistry.getTargetContext();
    }

    static Database provideDatabase() {
        return new Database(provideContext(), null);
    }
}

File: storage/src/androidTest/java/pl/expensive/storage/Injector.kt
package pl.expensive.storage

import android.content.Context
import android.content.SharedPreferences
import android.support.test.InstrumentationRegistry

internal object Injector {
    fun provideContext(): Context {
        return InstrumentationRegistry.getTargetContext()
    }

    fun provideDatabase(): Database {
        return Database(provideContext(), provideCurrencyRepository(), null)
    }

    fun provideCurrencyRepository(): CurrencyRepository = CurrencyRepository(provideKeyValueStorage())
    fun provideKeyValueStorage(): SharedPreferences = provideContext().getSharedPreferences("test_sp", 0)
}
